### ISSUE TITLE
[acme.dnsChallenge]   not working with gandiv5: No or incorrect TXT record

### ISSUE THREAD

**Issue by retog** [containous/traefik#3445]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

After a change of config I noticed the client no longer gets an accepted certificate. Switching back to the previously working config didn't solve the problem, neither did downgrading to 1.6.2 or 1.6.1..

### What did you expect to see?

A working letsencrypt certificate

### What did you see instead?

No certificate is generated, but an error appears in the log (see below)

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.6.3
Codename:     tetedemoine
Go version:   go1.10.2
Built:        2018-06-05_03:29:01PM
OS/Arch:      linux/amd64
```

```
Same results with 1.6.1, 1.6.2 and 1.6.3
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
[acme]
 ...
  caServer = "https://acme-v02.api.letsencrypt.org/directory"

[acme.dnsChallenge]
  provider = "gandiv5"
  delayBeforeCheck = 0
...

[[acme.domains]]
  main = "*.example.org"
  sans = ["example.org"]

[[acme.domains]]
  main = "*.example.guru"
  sans = ["example.guru"]

[[acme.domains]]
  main = "service.example.com"
```

The results are the same using:

```toml
caServer = "https://acme-staging-v02.api.letsencrypt.org/directory"
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
6/6/2018 11:31:34 AMGandi DNS: Zone record deleted
6/6/2018 11:31:34 AMERRO[2018-06-06T09:31:34Z] Unable to obtain ACME certificate for domains "service.example.com" : cannot obtain certificates: acme: Error -> One or more domains had a problem:
6/6/2018 11:31:34 AM[service.example.com] acme: Error 400 - urn:ietf:params:acme:error:dns - DNS problem: NXDOMAIN looking up TXT for _acme-challenge.service.example.com
6/6/2018 11:31:34 AM
6/6/2018 11:32:01 AMGandi DNS: Zone record deleted
6/6/2018 11:32:01 AMERRO[2018-06-06T09:32:01Z] Unable to obtain ACME certificate for domains "*.example.guru,example.guru" : cannot obtain certificates: acme: Error -> One or more domains had a problem:
6/6/2018 11:32:01 AM[example.guru] acme: Error 403 - urn:ietf:params:acme:error:unauthorized - No TXT record found at _acme-challenge.example.guru
6/6/2018 11:32:01 AM
6/6/2018 11:32:35 AMGandi DNS: Zone record deleted
6/6/2018 11:32:36 AMGandi DNS: DNS Record Created
6/6/2018 11:33:21 AMGandi DNS: Zone record deleted
6/6/2018 11:33:21 AMERRO[2018-06-06T09:33:21Z] Unable to obtain ACME certificate for domains "*.example.org,example.org" : cannot obtain certificates: acme: Error -> One or more domains had a problem:
6/6/2018 11:33:21 AM[example.org] acme: Error 403 - urn:ietf:params:acme:error:unauthorized - Incorrect TXT record "nYhYDZD1ENYNTDy4_eQ5bP7vhhTVY5xWKrv_8pF3r2U" found at _acme-challenge.example.org

```

**Comment by emilevauge** [comment]
Any idea why it was working before your config change? There is no real reason for that :thinking:

**Comment by retog** [comment]
The problem disappeared again like it appeared, without me doing anything I'm having valid certs again. I hope it will stay that way. Thanks for the great product and for looking into this issue.

**Comment by retog** [comment]
And the problem appeared again, wildcard certs seem to fail with "Incorrect TXT record" more often than not.

**Comment by retog** [comment]
The symptoms seem to be the sme with cloudfare: https://serverfault.com/questions/916427/traefik-k8s-lets-encrypt-wildcard-ssl-cloudflare-issue

**Comment by retog** [comment]
I found a work around:

Instead of:
```
[[acme.domains]]
  main = "*.example.org"
  sans = ["example.org"]
```

I use:

```
[[acme.domains]]
  main = "*.example.org"

[[acme.domains]]
  main = "example.org"

```

**Comment by nmengin** [comment]
Hello @retog,

Many thanks for your feedback.

It seems to be related to a problem when you ask ACME to generate DNS challenge for both wildcard certificate and its root domain as mentionned in [the documentation](https://docs.traefik.io/configuration/acme/#wildcard-domains).

The tests we made with GandiV5 have generated successfully the certificates but, it looks to be flacky.
The problem is about the TXT cache timeout in DNS providers: Let's Encrypt provides the same TXT record name for both the wildcard ad the root domain but if the cache is not clear between the challenges, ACME cannot get all the values.

Solving this weird issue, or trying to help LEGO team (LEGO is library we use to create ACME challenges) to solve it, is in our agenda

**Comment by ntnn** [comment]
Additional to separating the domain names like @retog suggessted I only got it working with gandiv5 by setting `delayBeforeCheck` to at least 90s. Lower than that caused the error described above.

**Comment by nmengin** [comment]
Hello @retog  , @ntnn .

Can you try the new Træfik release candidate (1.7-RC3) please?

Indeed, this RC contains the bugfix I made (#3675).

For information, I have discussed with Let's Encrypt developer about the problem in https://github.com/letsencrypt/boulder/issues/3572.

**Comment by nmengin** [comment]
Closed by #3675
